tests/phpunit/tests/date/currentTime.php

```diff
@@ -157,4 +157,36 @@ public function test_should_work_with_deprecated_timezone() {
 		$this->assertSame( gmdate( $format ), $current_time_gmt, 'The dates should be equal [3]' );
 		$this->assertSame( $datetime->format( $format ), $current_time, 'The dates should be equal [4]' );
 	}
+
+	/**
+	 * @ticket 57998
+	 *
+	 * @dataProvider data_partial_hour_timezones_with_timestamp
+	 *
+	 * @param float $partial_hour Partial hour GMT offset to test.
+	 */
+	public function test_partial_hour_timezones_with_timestamp( $partial_hour ) {
+		update_option( 'gmt_offset', $partial_hour );
+
+		$expected = time() + (int) ( $partial_hour * HOUR_IN_SECONDS );
```

Comment:
Dynamic calculation causes false negatives when the clock moves into a different second between calculating expected and actual value. I vaguely remember some tests needed to be fuzzied to match within a second, rather than exactly.

Reply:
@Rarst Just confirming: WordPress uses +3:00 for Riyadh, and doesn't consider Riyadh 87-89 timezones valid, right?

Reply:
I am not familiar with that issue, WordPress uses time zones exactly as provided by PHP it's running on. There is no data or special handling for any time zones in WP itself (other than older system of unnamed numeric offsets and some localized WP versions can prefer numeric zone default over a named ones for technical/historical reasons).

Reply:
Thanks! Seems fine then, PHP only has `Asia/Riyadh` [Ref](https://www.php.net/manual/en/timezones.asia.php).